src/hooks/useTopTokens.ts

```diff
@@ -0,0 +1,158 @@
+    </svg>`,
+    price: {
+      [SupportedChainId.ARBITRUM_ONE]: {
+        [TimePeriod.hour]: 0.66543,
+        [TimePeriod.day]: 0.67,
+        [TimePeriod.week]: 0.66543,
+        [TimePeriod.month]: 0.9,
+      },
+      [SupportedChainId.MAINNET]: {
+        [TimePeriod.hour]: 0.66543,
+        [TimePeriod.day]: 0.67,
+        [TimePeriod.week]: 0.66543,
+        [TimePeriod.month]: 0.9,
+      },
+      [SupportedChainId.OPTIMISM]: {
+        [TimePeriod.hour]: 0.66543,
+        [TimePeriod.day]: 0.67,
+        [TimePeriod.week]: 0.66543,
+        [TimePeriod.month]: 0.9,
+      },
+    },
+    volume: {
+      [SupportedChainId.ARBITRUM_ONE]: {
+        [TimePeriod.hour]: 1_000,
+        [TimePeriod.day]: 24_000,
+        [TimePeriod.week]: 175_000,
+        [TimePeriod.month]: 700_000,
+      },
+      [SupportedChainId.MAINNET]: {
+        [TimePeriod.hour]: 5_000,
+        [TimePeriod.day]: 100_000,
+        [TimePeriod.week]: 800_000,
+        [TimePeriod.month]: 4_920_000,
+      },
+      [SupportedChainId.OPTIMISM]: {
+        [TimePeriod.hour]: 2_000,
+        [TimePeriod.day]: 48_000,
+        [TimePeriod.week]: 800_000,
+        [TimePeriod.month]: 2_100_000,
+      },
+    },
+  },
+}
+
+const useTopTokens = (): UseTopTokensResult => {
+  const [data, setData] = useState<TokenData | null>(null)
+  const [error, setError] = useState<string | null>(null)
+  const [loading, setLoading] = useState(false)
+
+  const fetchTopTokens = async (): Promise<TokenData | void> => {
+    const waitRandom = (min: number, max: number): Promise<void> =>
+      new Promise((resolve) => setTimeout(resolve, min + Math.round(Math.random() * Math.max(0, max - min))))
+    try {
+      setLoading(true)
+      setError(null)
+      await waitRandom(250, 2000)
+      if (Math.random() < 0.05) {
+        throw new Error('fake error')
+      }
+      return FAKE_TOP_TOKENS_RESULT
+    } catch (e) {
+      setError('something went wrong')
+    } finally {
+      setLoading(false)
+    }
+  }
+
+  useEffect(() => {
+    setLoading(true)
+    setError(null)
+    fetchTopTokens()
+      .then((data) => {
+        if (data) setData(data)
+      })
+      .catch((e) => setError(e))
+      .finally(() => setLoading(false))
+  }, [])
+
+  return { data, error, loading }
```

Comment:
is there a way to refetch the data exposed?

Reply:
i think you could just have a `refetch` function passed as part of the hook

Reply:
`pollingInterval`